src/stepper-motor-element.ts

```diff
@@ -1,177 +1,126 @@
+  textSize: number;     // Text size showing units etc
+};
 
 @customElement('wokwi-stepper-motor')
 export class StepperMotorElement extends LitElement {
   @property() angle = 0;
   @property() arrow = '';
   @property() value = '';
   @property() units = '';
+  @property() size : 8 | 11 | 14 | 17 | 23 | 34=17; // valid sizes are 8, 11, 14, 17, 21, 24. 8 is the default at 0.8"
+
+  get pinInfo(): ElementPin[] {
+    const spec = this.nemaSpecMap[this.size];
+
+    // these offsets match the transform in renderFace
+    const xOff = (spec.frameSize / 2 - 3.75) * mmToPix;
+    const yOff = (spec.frameSize + 5) * mmToPix;
+
+    const pi = [
+      { name: 'A+', y: yOff, x: xOff, number: 1, signals: [] },
+      { name: 'A-', y: yOff, x: xOff + 2.54 * mmToPix, number: 2, signals: [] },
+      { name: 'B+', y: yOff, x: xOff + 5.08 * mmToPix, number: 3, signals: [] },
+      { name: 'B-', y: yOff, x: xOff + 7.62 * mmToPix, number: 4, signals: [] },
+    ];
+
+    return pi; 
+  }
+
+  readonly nemaSpecMap : {[key:string] : NemaSpec } = {
+    "8":  { "id": 8, "frameSize": 20.4, "holeRadius": 0.5, "shaftRadius": 3.5, "cornerRadius": 2.5, "cornerOffset": 2.5, "bodyRadius": 7.7, "textSize":8 },
+    "11": { "id": 11, "frameSize": 28.2, "holeRadius": 1.25, "shaftRadius": 5, "cornerRadius": 2.5, "cornerOffset": 2.5, "bodyRadius": 11, "textSize":8 },
+    "14": { "id": 14, "frameSize": 35.2, "holeRadius": 1.5, "shaftRadius": 5, "cornerRadius": 4.5, "cornerOffset": 4.5, "bodyRadius": 11, "textSize":10 },
+    "17": { "id": 17, "frameSize": 42.3, "holeRadius": 1.5, "shaftRadius": 5, "cornerRadius": 5, "cornerOffset": 5.5, "bodyRadius": 14, "textSize":10 },
+    "23": { "id": 23, "frameSize": 57.3, "holeRadius": 2.5, "shaftRadius": 6.35, "cornerRadius": 5, "cornerOffset": 5.5, "bodyRadius": 19.5, "textSize":16 },
+    "34": { "id": 34, "frameSize": 86, "holeRadius": 3.25, "shaftRadius": 14, "cornerRadius": 3.25, "cornerOffset": 8.4, "bodyRadius": 36.5, "textSize":16 },
+  }
 
-  readonly pinInfo: ElementPin[] = [
-    { name: 'A+', y: 237, x: 108, number: 1, signals: [] },
-    { name: 'A-', y: 237, x: 98, number: 2, signals: [] },
-    { name: 'B+', y: 237, x: 117, number: 3, signals: [] },
-    { name: 'B-', y: 237, x: 127, number: 4, signals: [] },
-  ];
+  render() { 
+    const spec = this.nemaSpecMap[this.size];
 
-  render() {
-    const { arrow } = this;
+    // local vars    const ratio = 1; //100 / spec.frameSize;
+    const cornerRadius = spec.cornerRadius;
+    const holeRadius = spec.holeRadius;
+    const shaftRadius = spec.shaftRadius;
+    const frameSize = spec.frameSize;
+    const cornerOffset = spec.cornerOffset;
+    const bodyRadius = spec.bodyRadius;
 
-    return html`
-      <svg
-        width="57mm"
-        height="63.349mm"
-        version="1.1"
-        viewBox="0 0 57 63.349"
-        xmlns="http://www.w3.org/2000/svg"
-        xmlns:xlink="http://www.w3.org/1999/xlink"
+    return html`<svg
+      width="100mm"
```

Comment:
SVG size should match the actual size of the graphics. Otherwise, Wokwi will draw the selection rectangle too big when you click on the element.

Reply:
are you suggesting that I should vary the generated svg to match the size of the component?
I can do that